src/ambianic/pipeline/ai/fall_detect.py

```diff
@@ -171,34 +185,41 @@ def fall_detect(self, image=None):
                 if not self._prev_vals or lapse > self.max_time_between_frames:
                     log.info("No recent pose to compare to. Will save this frame pose for subsequent comparison.")
                 else:
-                    left_angle = 0
-                    left_body_line_detected = len(pose_vals_list[0]) == 2
-                    if left_body_line_detected and (self._prev_vals[0] is not None):
-                        try:
-                            temp_left_point = [[self._prev_vals[0][0], self._prev_vals[0][1]], [pose_vals_list[0][0], pose_vals_list[0][1]]]
-                            left_angle = self.calculate_angle(temp_left_point)
-                            log.info("Left shoulder-hip angle: %r", left_angle)
-                        except IndexError as e:
-                            log.exception("Error while calculating angle: %r", e)
-                    right_angle = 0
-                    right_body_line_detected = len(pose_vals_list[1]) == 2
-                    if right_body_line_detected and (self._prev_vals[1] is not None):
-                        try:
-                            temp_right_point = [[self._prev_vals[1][0], self._prev_vals[1][1]], [pose_vals_list[1][0], pose_vals_list[1][1]]]
-                            right_angle = self.calculate_angle(temp_right_point)
-                            log.info("Right shoulder-hip angle: %r", left_angle)
-                        except IndexError as e:
-                            log.exception("Error while calculating angle: %r", e)
-
-                    angle_change = max(left_angle, right_angle)
-                    if angle_change > self._fall_factor:
-                        # insert a box that covers the whole image as a workaround
-                        # to meet the expected format of the save_detections element
-                        box = [0, 0, 1, 1]
-                        inference_result.append(('FALL', pose.score, box, angle_change ))
-                        log.info("Fall detected: %r", inference_result)
+                    if (left_angle_with_yaxis and self._prev_left_angle_with_yaxis and left_angle_with_yaxis > self._prev_left_angle_with_yaxis) or (rigth_angle_with_yaxis and self._prev_right_angle_with_yaxis and rigth_angle_with_yaxis > self._prev_right_angle_with_yaxis):
```

Comment:
@bhavikapanara I think this logic is correct, but as a best practice, we should avoid such long and convoluted conditionals.

Please extract this conditional into a separate method (maybe is_body_line_motion_downward() ) and document its purpose for better code maintainability. Thank you!

Reply:
We will respectively need to add a few test cases to cover the various scenarious.

Reply:
> @bhavikapanara I think this logic is correct, but as a best practice, we should avoid such long and convoluted conditionals.
> 
> Please extract this conditional into a separate method (maybe is_body_line_motion_downward() ) and document its purpose for better code maintainability. Thank you!

Done

Reply:
> We will respectively need to add a few test cases to cover the various scenarious.

working to get images for test cases to cover this scenario.

Reply:
> working to get images for test cases to cover this scenario.

OK, please use the new images contributed by our users.